src/commands/cmd_key.cc

```diff
@@ -553,6 +553,26 @@ class CommandSort : public Commander {
   SortArgument sort_argument_;
 };
 
+class CommandKMetadata : public Commander {
+ public:
+  Status Execute(engine::Context &ctx, Server *srv, Connection *conn, std::string *output) override {
+    redis::Database redis(srv->storage, conn->GetNamespace());
+    std::string &key = args_[1];
+    std::string nskey = redis.AppendNamespacePrefix(key);
+
+    Metadata metadata(kRedisNone, false);
+    auto s = redis.GetMetadata(ctx, RedisTypes::All(), nskey, &metadata);
+    if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
+
+    *output = conn->Map({{redis::BulkString("type"), redis::BulkString(RedisTypeNames[metadata.Type()])},
+                         {redis::BulkString("size"), redis::Integer(metadata.size)},
+                         {redis::BulkString("ttl"), redis::Integer(metadata.expire)},
+                         {redis::BulkString("flags"), redis::Integer(metadata.flags)},
+                         {redis::BulkString("version"), redis::Integer(metadata.version)}});
```

Comment:
if `metadata.IsSingleKVType()` is true, we can remove `size` and `version` here since it's useless.

Reply:
thanks for the comment, it is true that those fields have no importance for singleKV, but should make easier to unmarshall the cmd response if we keep all the fields, wdyt?

Reply:
I think we should not give users useless information, even if it can be easy to decode. IMHO you can set a default value and use the default value if the field is missing.

Reply:
@PragmaTwice For the sideloading use case when IsSingleKVType() you will need version to perform the key update. So the only "useless" field would be size in that case, am I wrong?

Reply:
addressed via 10ced99e1695e9abef90a7deb8e21e1b00b77602